x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/CcsSlowLogRestIT.java

```diff
@@ -0,0 +1,358 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.remotecluster;
+
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.common.io.Streams;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.test.cluster.ElasticsearchCluster;
+import org.elasticsearch.test.cluster.LogType;
+import org.elasticsearch.test.rest.ObjectPath;
+import org.elasticsearch.xcontent.XContentType;
+import org.junit.ClassRule;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasKey;
+import static org.hamcrest.Matchers.notNullValue;
+
+/**
+ * Integration test for verifying that slow log authentication context contains
+ * the correct user information for cross-cluster access scenarios.
+ *
+ * This test verifies that when cross-cluster searches are performed, the slow logs
+ * on the fulfilling cluster contain the authentication context of the ORIGINAL user
+ * from the querying cluster, not just the cross-cluster access API key.
+ *
+ * The key verification is that slow logs should show:
+ * - user.name: The actual user from the querying cluster (e.g., "slow_log_test_user")
+ * - user.realm: The realm of the original user on the querying cluster (e.g., "default_native")
+ * - For run-as: Both authenticating and effective users from querying cluster
+ */
+public class CcsSlowLogRestIT extends AbstractRemoteClusterSecurityTestCase {
```

Comment:
Naming nit: `RemoteClusterSecuritySlowLogRestIT` better matches the other test case in the suite.